extensions/pkg/predicate/predicate.go

```diff
@@ -173,7 +173,7 @@ func LastOperationNotSuccessful() predicate.Predicate {
 		}
 
 		lastOp := acc.GetExtensionStatus().GetLastOperation()
-		return lastOp == nil ||
+		return lastOp != nil &&
```

Comment:
I'm not sure about all the consequences of this. Looping in @timuthy - WDYT?

Reply:
Good point. I think it is necessary to cover cases in which the last operation is not set at all, e.g. if one controller instance stops operation very early (maybe because the pod is evicted or it crashed) and the other starting instance must begin with the reconciliation again. The change above will prevent this in the future. What's the motivation for that change at all?

Reply:
If the pod crashes before setting the operation, Gardenlet will treat the infra as unsuccessful and annotate the object for reconciliation again right? The motivation was that when you create an Infrastructure object without annotations it was reconciled anyway, but we need to update its status with the state from the ShootState before starting the reconciliation. Because of this predicate the controller starts reconciliation, and we thought this check might not be correct because it states "LastOperationNotSuccessful", which in case of nil returns true. Anyway I am already adding new annotation to handle the wait-for-state scenario and @vlvasilev will prob revert this.

Reply:
> The motivation was that when you create an Infrastructure object without annotations it was reconciled anyway

This is a valid point - indeed, when the operation annotation is not ignored then this should not happen.

Reply:
> If the pod crashes before setting the operation, Gardenlet will treat the infra as unsuccessful and annotate the object for reconciliation again right?

This only happens after the Gardener flow step times out. It'll then retry it again, i.e. sets the operation annotation during the next reconciliation flow.

Reply:
@vlvasilev @swilen-iwanow what's the status?

Reply:
We are going to revert this, but still have to implement the wait-for-migrate state for the extension controllers. Should we do this in this PR or open a separate.

Reply:
Do you mean for the restoration case? Probably it should be this PR, or?